crates/node/src/catchup.rs

```diff
@@ -58,7 +65,7 @@ impl Node {
 
         let application_id = context.application_id.clone();
 
-        if request.application_id.is_none() || application_id == request.application_id.unwrap() {
+        if request.application_id.is_none() || application_id != request.application_id.unwrap() {
```

Comment:
either of these get the same result without invoking `unwrap`

```suggestion
        if matches!(request.application_id, Some(id) if application_id == id) {}
```

```suggestion
        if Some(application_id) == request.application_id {}
```

Reply:
> With second approach I have to clone application_id that is being wrapped in Some

nothing a little `as_ref` won't fix

```suggestion
        if Some(&application_id) == request.application_id.as_ref() {}
```

> And with the first approach I also have to clone application_id

weird, works on my end